index.bs

```diff
@@ -9024,6 +9070,386 @@ worker=] algorithm:
+        storage.CookieFilter = {
+          ? name: text,
+          ? value: network.BytesValue,
+          ? domain: text,
+          ? path: text,
+          ? size: js-uint,
+          ? httpOnly: bool,
+          ? secure: bool,
+          ? sameSite: network.SameSite,
+          ? expiry: js-uint,
+          Extensible,
+        }
+
+        storage.BrowsingContextPartitionDescriptor = {
+          type: "context",
+          context: browsingContext.BrowsingContext
+        }
+
+        storage.StorageKeyPartitionDescriptor = {
+          type: "storageKey",
+          ? userContext: text,
+          ? sourceOrigin: text,
+          Extensible,
+        }
+
+        storage.PartitionDescriptor = (
+          storage.BrowsingContextPartitionDescriptor /
+          storage.StorageKeyPartitionDescriptor
+        )
+
+        storage.GetCookiesParameters = {
+          ? filter: storage.CookieFilter,
+          ? partition: storage.PartitionDescriptor,
+        }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      storage.GetCookiesResult = {
+        cookies: [*network.Cookie],
+        partitionKey: storage.PartitionKey,
+      }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for storage.getCookies">
+The [=remote end steps=] with <var ignore>session</var> and |command parameters| are:
+
+1. Let |filter| be the value of the <code>filter</code> field of |command parameters|
+  if it is present or an empty [=/map=] if it isn't.
+
+1. Let |partition spec| be the value of the <code>partition</code> field of
+  |command parameters| if it is present or null if it isn't.
+
+1. Let |partition key| be the result of [=trying=] to [=expand a storage partition spec|expand=] |partition spec|.
+
+1. Let |store| be the result of [=trying=] to [=get the cookie store=] with
+  |partition key|.
+
+1. Let |cookies| be the result of [=get matching cookies=] with |store| and |filter|.
+
+1. Let |body| be a [=/map=] matching the <code>storage.GetCookiesResult</code> production,
+  with the <code>cookies</code> field set to |cookies| and the <code>partitionKey</code>
+  field set to |partition key|.
+
+1. Return [=success=] with data |body|.
+
+</div>
+
+#### The storage.setCookie Command ####  {#command-storage-setCookie}
+
+The <dfn export for=commands>storage.setCookie</dfn> command creates a new [=cookie=] in a cookie store, replacing any cookie in that store which matches according to [[COOKIES]].
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+        storage.SetCookie = (
+          method: "storage.setCookie",
+          params: storage.SetCookieParameters,
+        )
+
+        <!--
+        Modifications to this definition should be reflected in
+        `network.Cookie`, `network.SetCookieHeader`, and `storage.CookieFilter`.
+        -->
+        storage.PartialCookie = {
+          name: text,
+          value: network.BytesValue,
+          domain: text,
```

Comment:
Do we want to duplicate domain, if the `browsingContext` is provided?

Reply:
This is the domain the cookie will be sent to, not the domain of the page the cookie will be sent from, so I'm not sure what the duplication is?

Reply:
Oh, my bad. Thanks!

Reply:
Does it make sense to set it optional if the expected domain to be sent to is the same as the one in partition descriptor?

Reply:
That feels like too much magic to me.